packages/shared/components/Button.svelte

```diff
@@ -1,532 +1,261 @@
         }
-        &:not(.secondary) {
-            span {
-                @apply font-bold;
-            }
+
+        &:hover {
+            @apply bg-#{$color}-600;
         }
-        &:not(.with-icon) {
-            &:hover,
-            &:focus {
-                @apply bg-blue-600;
-            }
-            &:active {
-                @apply bg-blue-700;
-            }
-            &.caution {
-                @apply bg-yellow-600;
-                min-width: 100px;
-                span {
-                    @apply text-white;
-                }
-                &:hover {
-                    @apply bg-yellow-700;
-                }
-                &:active,
-                &:focus {
-                    @apply bg-yellow-800;
-                }
-                &:disabled {
-                    @apply pointer-events-none;
-                    @apply bg-gray-200;
-                    span {
-                        @apply text-gray-500;
-                    }
-                    &.darkmode {
-                        @apply bg-gray-700;
-                        @apply bg-opacity-10;
-                        span {
-                            @apply text-gray-700;
-                        }
-                    }
-                }
-            }
-            &.warning {
-                @apply bg-red-500;
-                min-width: 100px;
-                span {
-                    @apply text-white;
-                }
-                &:hover {
-                    @apply bg-red-600;
-                }
-                &:active,
-                &:focus {
-                    @apply bg-red-700;
-                }
-                &:disabled {
-                    @apply pointer-events-none;
-                    @apply bg-gray-200;
-                    span {
-                        @apply text-gray-500;
-                    }
-                    &.darkmode {
-                        @apply bg-gray-700;
-                        @apply bg-opacity-10;
-                        span {
-                            @apply text-gray-700;
-                        }
-                    }
-                }
-            }
-            &.medium {
-                @apply pt-1.5;
-                @apply pb-2.5;
-            }
+
+        &:active,
+        &:focus {
+            @apply bg-#{$color}-700;
+            @apply ring-4;
+            @apply ring-#{$color}-400;
+            @apply ring-opacity-20;
         }
-        &.secondary {
+
+        &.outline {
+            @apply bg-white;
+            @apply text-#{$color}-500;
             @apply border;
             @apply border-solid;
             @apply border-gray-300;
-            @apply bg-white;
-            span {
-                @apply text-blue-500;
-            }
+
             &:hover {
-                @apply bg-blue-50;
-                @apply border-blue-200;
+                @apply bg-#{$color}-50;
+                @apply border-#{$color}-200;
```

Comment:
maybe for the time being, until we upgrade tailwind to v3, its worth adding to the tailwind safelist `/^border-/` to avoid tailwind purging these classes? https://v2.tailwindcss.com/docs/optimizing-for-production#writing-purgeable-html